### PR TITLE
test: initial tests for WordReader

### PR DESCRIPTION
**Pull request by magnopus-swifty**
Initial set of tests for `WordReader`.

With these changes we're at 95%+ test coverage for the `WordReader` class, although I think that's misleading because that last 5% is where the obscure edge cases live.

### CODE REVIEW

File: Tools/WrapperGenerator/tests/test_word_reader.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    def test_peek_char_and_skip_char(self):
+        w = WordReader("abc")
+        self.assertEqual(w.peek_char(), "a")
+        w.skip_char()
+        self.assertEqual(w.peek_char(), "b")
+        w.skip_char(2)
+        self.assertIsNone(w.peek_char())
+
+    def test_skip(self):
+        w = WordReader("a\nb\nc")
```
Comment (MAG-ChristopherAtkinson) [inline Tools/WrapperGenerator/tests/test_word_reader.py:45]: [Question] same as above ^

File: Tools/WrapperGenerator/tests/test_word_reader.py
Comment (MAG-ChristopherAtkinson) [inline Tools/WrapperGenerator/tests/test_word_reader.py:1]: [Question] I wonder if there is any value in testing against a utf-16 characters, not sure off the top of my head how Python or our implementation handles string encoding

File: Tools/WrapperGenerator/tests/test_word_reader.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        w.skip(2)  # skips 'a' and '\n'
+        self.assertEqual(w.current_line, 2)
+        self.assertEqual(w.peek_char(), "b")
+        w.skip(2)  # skips 'b' and '\n'
+        self.assertEqual(w.current_line, 3)
+        self.assertEqual(w.peek_char(), "c")
+
+    def test_skip_line(self):
+        w = WordReader("foo\nbar\nbaz")
```
Comment (MAG-ChristopherAtkinson) [inline Tools/WrapperGenerator/tests/test_word_reader.py:54]: [Question] same as above ^

File: Tools/WrapperGenerator/tests/test_word_reader.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        w.skip(4)
+        self.assertEqual(w.find_next_of({"e", "f"}), "e")
+
+    def test_find_next_of_raises_index_error(self):
+        w = WordReader("abc")
+        with self.assertRaises(IndexError):
+            w.find_next_of({"z"})
+
+    def test_next_word_handles_only_whitespace(self):
+        w = WordReader("   \t\n  ")
```
Comment (MAG-ChristopherAtkinson) [inline Tools/WrapperGenerator/tests/test_word_reader.py:77]: [Question] same as above ^

File: Tools/WrapperGenerator/tests/test_word_reader.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+class TestWordReader(unittest.TestCase):
+    def test_next_word_basic(self):
+        w = WordReader("Hello, world!")
+        self.assertEqual(w.next_word(), "Hello")
+        self.assertEqual(w.next_word(), ",")
+        self.assertEqual(w.next_word(), "world!")
+        self.assertIsNone(w.next_word())
+
+    def test_skip_whitespace_and_peek_char(self):
+        w = WordReader("   \t\n  Meow")
```
Comment (MAG-ChristopherAtkinson) [inline Tools/WrapperGenerator/tests/test_word_reader.py:13]: [Question] Should we be including a carriage return here as well? \r\n

**Review by MAG-ElliotMorris — APPROVED** [review]
Looks good, although not too fussed about WordReader. It's always seemed solid, and I doubt it survives/needs changing into whatever the next implementation of the parser is. (tbh, I'm hopeful the python language won't survive, dynamically typed languages are bad at this sort of thing imo)

File: Tools/WrapperGenerator/tests/test_word_reader.py
Comment (magnopus-swifty) [reply Tools/WrapperGenerator/tests/test_word_reader.py:1]: I think we'd handle that downstream. Assuming you open the file with the correct encoding then we just pass the tokens on and it's up to the parser to decide if it cares (right now there's not much validation).

File: Tools/WrapperGenerator/tests/test_word_reader.py (hunk 4 above)
Comment (magnopus-swifty) [reply Tools/WrapperGenerator/tests/test_word_reader.py:13]: Assuming you open the file as text (`open("...", "t")`) then it'll convert line endings so I think we can just hand off responsibility that way.

**Review by MAG-ChristopherAtkinson — APPROVED** [review]
